### ISSUE TITLE
HUB320 support

### ISSUE THREAD

**Issue by davidkessler-ch** [hzeller/rpi-rgb-led-matrix#1618]
So all the matrices listed use HUB75, but for bigger matrices there is the similar HUB320. Are there plans to support that?

I have a 256*128 pixel matrix with HUB320, a 1/64S see [here](https://de.aliexpress.com/item/1005004995908838.html)

**Comment by wata-net** [comment]
There is nothing the library can do about it.
All you have to do is make a conversion cable.
![1](https://github.com/hzeller/rpi-rgb-led-matrix/assets/122246369/24f4ba41-b7f4-4bdc-9480-b55c26eba86e)

**Comment by Keniis0712** [comment]
> There is nothing the library can do about it. All you have to do is make a conversion cable. ![1](https://private-user-images.githubusercontent.com/122246369/298236731-24f4ba41-b7f4-4bdc-9480-b55c26eba86e.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk3MzEyODksIm5iZiI6MTcwOTczMDk4OSwicGF0aCI6Ii8xMjIyNDYzNjkvMjk4MjM2NzMxLTI0ZjRiYTQxLWI3ZjQtNGJkYy05NDgwLWI1NWMyNmViYTg2ZS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMzA2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDMwNlQxMzE2MjlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYTMzZGNlMTM4NTJlZTI1OTk0ODU1MTQwNjI0MGNlNWRhNzZhYTUzMGM3ZTE4M2UzYTczMjBiZjE1ODNlZWZlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.8Zz_IzX-DetGVMCYLT0NXJ4G82detOyBw3mQ7SMSM-Q) However, most HUB320 LED panels use driver chips with built-in SRAM. This library does not support driver chips with built-in SRAM. (This is not related to the HUB320 issue.)

I have a special driver chips withnot built-in SRAM and I only find the library for HUB75. How can I use this kind of library to control a HUB320 screen?

**Comment by krambs** [comment]
Anyone able to get this library working with HUB320 and 256x128 yet? Thanks!

**Comment by marcmerlin** [comment]
@krambs  it's a new setup, new chip, new way to drive it and new cable.  Unless you write it yourself or pay someone to write it, it's probably not likely to happen unfortunately.
Also, note that panels that do not have an output plug, cannot be chained, so you'll really want an FPGA card with lots of output cables to run a bunch of those.

**Comment by sub-zerohm** [comment]
I've looked into this myself, if you lookup Kevtris on YT he has a video of how he reverse engineered one to work with jumpers soldered the NT Mini's NES cartridge pins (with a cart adapter as to not ruin the actual NT) and the open FPGA with his own custom gameboy core....but those are 26-pin connectors, and the pi only has 40 pins.  20 of which are actually not ground pins, but that means you need 21 pins to get one panel working, leaving no possibility for more than a single panel since they're designed to not be chained.

I want to port the library to an NES emulator but I don't think the pi4 can handle it, waiting patiently for the library to come to the pi5 🙌 hzeller is the man

**Comment by sub-zerohm** [comment]
Actually was wondering if these have sram.  Can't tell by looking.  I ordered them direct from the supplier

![PXL_20241228_053527503 MP](https://github.com/user-attachments/assets/5a761ceb-3a6b-4622-ac3b-94e5796550d2)

**Comment by rxchit** [comment]
I'm also working with one of the panels with HUB320 connectors but the pinout is slightly different, can you guys point in any direction to get it working with pi4/5![image](https://github.com/user-attachments/assets/9ef030df-89ec-402b-a10a-cbb7295fcad5)

**Comment by sonycman** [comment]
**>@sub-zerohm**
> Actually was wondering if these have sram. Can't tell by looking. I ordered them direct from the supplier. If not I could run jumpers off the raspberry pi pins but would need to mess with the HAT type. They're 1.25 pitch panels, 256x128 resolution. 2 of them would make a perfect retro game screen
> 
> It looks like they're using address type c which basically clocks each color twice to achieve the scan rate, but refresh rate that the panel can support is cut in half. I chained two 128x64 panels that use this same addressing type together and you have to go kind of basic on the content or else it will flicker
>
So is there any luck handling these large 256 x 128 panels?
Could you, please, post some quality photos of your panel, so i could see the chips its using?
Would be appreciated a lot!

**Comment by sonycman** [comment]
**@rxchit**
> I'm also working with one of the panels with HUB320 connectors but the pinout is slightly different, can you guys point in any direction to get it working with pi4/5!

Hi! Would you be so kind to also repost a better quality photo of panel PCB?
Guys, what a potato cameras do you use!

**Comment by sonycman** [comment]
@wata-net

Regarding the panels of 256 x 128 pixels (320 x 160mm), HUB320.
Asked vendors on aliexpress, and they have all the different chips:
1: DP3364S
2: ICN1065
3: [FM6363C and FM6565C](https://www.eevblog.com/forum/projects/datasheet-for-led-drivers-fm6363c-and-fm6565c/)

The first have good datasheet, but without registers description, of course.
For the second i've found nothing.
The third - FM6363 used successfully somewhere here and no problem, dont know about FM6565 though...

But chinese vendors could not guarantee exact chipset with all the panels they sell (its random), so there is no go for now for these big 256x128 panels, am i right?

**Comment by martodg** [comment]
Hi everyone,
I purchased a 256X128 P1.25 LED module with a HUB75E interface. The chips appear to be DP3364S, DP245D and RUL5158E
I bought the unit from AliExpress:
https://www.aliexpress.com/item/1005006086019360.html
Any idea on how to get the panel to display? I've used other panels in the past and have gotten them to work with variations of the led panel type, gpio slowdown, etc.
Look forward to a response.
Cheers.

**Comment by marcmerlin** [comment]
@martodg  why are you posting in a HUB320 bug when your panels are HUB75E?
It also does not look like you read the documentation
https://github.com/hzeller/rpi-rgb-led-matrix?tab=readme-ov-file#panels-supported
you will find your answer there that your panel does not look supported, and in the future please do not hijack an unrelated bug, please.

**Comment by martodg** [comment]
Sorry @marcmerlin 
I had no intention of hijacking a bug or upsetting anyone. There weren't may posts on 256X128 panels.
Thanks for the update

**Comment by marcmerlin** [comment]
@martodg  you're supposed to make a new issue, not to post in a random unrelated one.
But there aren't posts indeed because those panels aren't supported as per the README.  
As of right now, I have not seen a single 128 line panel work with the existing libs, and it's likely because without using PWM the refresh rate would be terrible, and PWM chips are completely different, and unsupported, until someone writes a driver for one of them. No one has offered to do so as of today.

**Comment by sonycman** [comment]
@martodg 
Hi. Successfully running similar panel (DP3364S + RUL5258 + HUB320) on custom hardware (FPGA).
Now just received same panel as yours - DP3364S + RUL5158 + HUB75E.
Seems almost the same both of them, except less data channels on latter, thus it might need higher clock or dual data rate for DP3364...

**Comment by martodg** [comment]
@sonycman 
That's fantastic news. Thanks for letting me know.
Can I ask what flags or drivers you have used to get the panel displaying? I tried as many variations, but couldn't get any pixels to light up. I'd really appreciate any feedback. I'm trying to get this panel working as a project I'm developing for my work.
Thanks again.

**Comment by sonycman** [comment]
@martodg 
I wrote my own driver (actually module in verilog) to run the panel directly by FPGA SoC.
DP3364S is very well documented, actually, and also thanks to additional help from panel seller.

On which hardware you have tried to get it working?

I dont know if this thread is good place to talk about...

**Comment by martodg** [comment]
@sonycman 
I have created a new issue:
https://github.com/hzeller/rpi-rgb-led-matrix/issues/1821
If you could post on that issue and assist, that would be great. Thanks again

**Comment by marcmerlin** [comment]
@davidkessler-ch  @wata-net  @sonycman @sub-zerohm  @martodg 
Can you summarize the current situation about HUB320?  First unless one of you is working on it, or has someone who will work on it for you, there is no one maintaining the lib who will reasonably working on this with the current state of affairs.
Next, I'm trying to understand a couple of things:
1)  Are any HUB320 panels able to run with the regular non PWM code we have (by modifying the code to support RGB3 and RGB4 pins and only a single channel instead of 3)?
2) Or are all HUB320 panels using PWM chips and as of right now the core library does not support them anyway, so this won't happen either.?

If I hear nothing back, I'll be tempted to close this as WONTFIX for now simply due to no one able to work on it, and even less likely to happen if PWM is required on top

**Comment by marcmerlin** [comment]
since we got no answer, I'm going to close this until someone wants to work on it and re-open it
